lib/logstash/plugin_mixins/scheduler.rb

```diff
@@ -0,0 +1,60 @@
+# encoding: utf-8
+
+require 'logstash/namespace'
+require 'logstash/plugin'
+
+require 'logstash/plugin_mixins/scheduler/rufus_impl'
+
+module LogStash
+  module PluginMixins
+    module Scheduler
+
+      # @param cron [String] cron-line
+      # @param opts [Hash] scheduler options
+      # @return scheduler instance
+      def start_cron_scheduler!(cron, opts = {}, &block)
+        unless block_given?
+          raise ArgumentError, 'missing block - worker task to execute'
+        end
+        scheduler = new_scheduler(opts)
+        scheduler.schedule_cron(cron, &block)
+        scheduler
+      end
+
+      %w(every at in interval).each do |type|
```

Comment:
unfortunately, the http-poller plugin [exported (using `schedule => `)](https://www.elastic.co/guide/en/logstash/current/plugins-inputs-http_poller.html#plugins-inputs-http_poller-schedule) all of Rufus' flexibility.

otherwise we mostly (only) use cron and interval in one other place.

the idea behind providing these helpers from the mixin is to make sure we use the `schedule_xxx` methods - which allow us to extend the `Rufus::Scheduler::Jobs` instance with behavior that sets the name for work threads.

Reply:
The `class_eval` block used to generate dynamically the `start_every_scheduler!` up to `start_interval_scheduler!` seems identical to `start_cron_scheduler!`, there is a reason why that is left out?

Reply:
nope - was planning to eventually merging them all to one template (unless I need some different defaults based on the scheduling type) but thought I leave it for now as it might help readability while reviewing ...